### ISSUE TITLE
HCC is deprecated

### ISSUE THREAD

**Issue by tomdeakin** [illuhad/hipSYCL#262]
Just an FYI whilst I was trawling through the ROCm GitHub page:

https://rocmdocs.amd.com/en/latest/Programming_Guides/Programming-Guides.html#

**Comment by illuhad** [comment]
yes, we have not been using hcc for quite some time now (we instead rely entirely on clang), but it's still mentioned in a couple of places (e.g. the readme or the deprecated `HIPSYCL_PLATFORM_HCC` macro).

**Comment by tomdeakin** [comment]
The trouble I ran into was building AMD's HIP still needs HCC. It looks like there [is a way to do it](https://github.com/ROCm-Developer-Tools/HIP/blob/master/INSTALL.md#hip-clang), but it's fiddly. The hipSYCL CMake checks for `hipcc`.

**Comment by illuhad** [comment]
yes, AMD's HIP runtime library is still somewhat based on hcc, that's true. But that's not really an hipSYCL issue, we just need to wait for AMD to get their stuff in order... The binary ROCm packages that we ship are based on AMD's aomp installation scripts, so as soon as they don't build hcc anymore, we also won't. So, from the hipSYCL side, I really don't see any direct dependencies on hcc, except for the references here and there that I had mentioned.
The short version of the story is that we switched away from hcc in time for AMD's original hcc deprecation timescale, which was summer 2019...

Yes, we check for `hipcc`, which can either use hcc, clang, or even nvcc under the hood I think. So that's also not a dependency on hcc. We only check for `hipcc` because there's no `find_package(rocm)` yet as far as I know, so we use `hipcc` as a proxy to check if there is a ROCm installation.

**Comment by tomdeakin** [comment]
Understood - although I do get confused how the ROCm/HIP/HCC stack is made available.

It's great you ship packages and install scripts. The pain is you have to have root which doesn't happen very often in the systems I use!! 😢

**Comment by illuhad** [comment]
It's not thoroughly tested, but in principle you can set the `INSTALL_PREFIX` environment variable to a user-writable location. Then you shouldn't need root for the installation scripts.

For your use case, the hipSYCL spack package might also be interesting. Unfortunately, there's no ROCm spack package yet (AMD said they are looking into it, so maybe there's hope), so this only works for CUDA and CPU backends.

**Comment by tomdeakin** [comment]
Leaving not for my future self when I run into similar issues, something to try is to delete the missing libraries ( in `ROCM_LINK_LINE` in `CMakeLists.txt`:
```
/usr/bin/ld: cannot find -lmcwamp
/usr/bin/ld: cannot find -lhc_am
```

**Comment by davidkennedydev** [comment]
@tomdeakin  Is it still to be done?

**Comment by illuhad** [comment]
We don't have any references to HCC anymore, apart from some compatibility macro.
